WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -77,8 +80,10 @@ public TransactionPreviewViewModel(Wallet wallet, TransactionInfo info, BitcoinA
 
 				if (x is ChangeAvoidanceSuggestionViewModel ca)
 				{
+					AddToUndoHistory();
 					_info.ChangelessCoins = ca.TransactionResult.SpentCoins;
-					UpdateTransaction(CurrentTransactionSummary, ca.TransactionResult);
+					_info.Amount = ca.TransactionResult.CalculateDestinationAmount();
```

Comment:
Why was this required?

Reply:
@soosr I tackled several existing bugs in this screen as part of this PR.

At first when I detected these, I thought they were caused by my code, but then I reproduced on master:

1 - If you apply privacy suggestions, and later change fee slider, you get the original transaction amount, because it wasn't updated in the `TransactionInfo` object

2 - Vice versa: if you update fee slider, and then apply privacy settings, you would get the old fee rate back

This PR attempts to fix all thos interactions between these screens/viewmodels

Reply:
> 1 - If you apply privacy suggestions, and later change fee slider, you get the original transaction amount, because it wasn't updated in the TransactionInfo object

This is intentional. Due to technical reasons after changing the transaction fee it should go back to the original transaction.

> 2 - Vice versa: if you update fee slider, and then apply privacy settings, you would get the old fee rate back

This seems to be a bug... If the transaction fee changes, it should start searching for new suggestions... that use the new transaction fee.

Reply:
@soosr should be fixed in latest commit

Reply:
> 2 - Vice versa: if you update fee slider, and then apply privacy settings, you would get the old fee rate back

@ichthus1604  I can't reproduce this on master